terraform/shared/scripts/client.sh

```diff
@@ -30,14 +30,23 @@ sudo cp $CONFIGDIR/consul_$CLOUD.service /etc/systemd/system/consul.service
 sudo systemctl start consul.service
 sleep 10
 
+# AWS CLI
+sudo DEBIAN_FRONTEND=noninteractive apt-get install -y python-pip
+sudo pip install awscli
+
 # Nomad
 
 ## Replace existing Nomad binary if remote file exists
 if [[ `wget -S --spider $NOMAD_BINARY  2>&1 | grep 'HTTP/1.1 200 OK'` ]]; then
-  curl -L $NOMAD_BINARY > nomad.zip
-  sudo unzip -o nomad.zip -d /usr/local/bin
-  sudo chmod 0755 /usr/local/bin/nomad
-  sudo chown root:root /usr/local/bin/nomad
+        curl -L $NOMAD_BINARY > nomad.zip
+        sudo unzip -o nomad.zip -d /usr/local/bin
+        sudo chmod 0755 /usr/local/bin/nomad
+        sudo chown root:root /usr/local/bin/nomad
+elif [[ "$(which aws) s3 cp $NOMAD_BINARY nomad.zip --dryrun" ]]; then
+        $(which aws) s3 cp $NOMAD_BINARY nomad.zip
```

Comment:
I would suggest reworking this a bit: I would start with S3 and match over BINARY path, and keep the common code outside the if statement, maybe something like:

```bash
if [[ "$NOMAD_BINARY" = s3:* ]]; then
  s3 cp $NOMAD_BINARY nomad.zip
elif [[ `wget -S --spider $NOMAD_BINARY  2>&1 | grep 'HTTP/1.1 200 OK'` ]]; then
  curl -L $NOMAD_BINARY > nomad.zip
fi

sudo unzip -o nomad.zip -d /usr/local/bin
sudo chmod 0755 /usr/local/bin/nomad
sudo chown root:root /usr/local/bin/nomad
```

I find the use of wget and curl here odd but it's not quite related to this code.

Reply:
Ah yes excellent suggestion. Thank you!